doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md

```diff
@@ -9,77 +9,75 @@ sort_idx: 10
-(See following [Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Device_management/provisionCredentials) call for more details). 
-This server-side profile will be used during endpoint registration. 
-Registration service is responsible for storage and queries of this information based on credentials or endpoint identifier.
-This feature may be useful in multiple use-cases. 
-For example, endpoint manufacturer may provision some secure information to the server-side endpoint profile based on security keys that are available only during manufacturing process.
+### Endpoint registration data
 
+Kaa platform user can use the endpoint credentials data to provision server-side profiles for new endpoints.
+This server-side profile will be used for endpoint registration.
+This can be useful when, for example, an endpoint manufacturer wants to provision sensitive information to the server-side endpoint profile based on the security keys that are only available during manufacturing process.
+For more information, see [Administration REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Device_management/provisionCredentials).
 
 ## Credentials service
 
-Kaa provides two credentials service implementations out-of-the box:
+Kaa provides two Credentials service implementations out of the box:
+
+- Trustful credentials service: default implementation that allows any endpoint to register and connect to Kaa cluster.
+
+- Internal credentials service: allows connection to Kaa cluster only to specified list of endpoints whose credentials were previously provisioned to Kaa server.
+For more information, see [Administration REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Device_management/provisionCredentials).
 
-1. Trustful credentials service -  default implementation that allows any endpoint to register and connect to Kaa cluster. (like the previous version of Kaa)
-2. Internal credentials service -  allows connecting with Kaa for specified list of endpoints whose credentials was previously provisioned with Kaa 
-[Admin REST API]({{root_url}}Programming-guide/Server-REST-APIs/#!/Device_management/provisionCredentials).
+### Custom implementation of Credentials service
 
-### Custom credentials service implementation
+Here is an example of a custom implementation of Credentials service:
 
-1. Create class which implements all method of 
-[CredentialsService interface](https://github.com/kaaproject/kaa/blob/1d429a30bb4b5206376b740bb21483929a881ace/server/node/src/main/java/org/kaaproject/kaa/server/node/service/credentials/CredentialsService.java)
+<ol start="1">
+<li markdown="1">
+Create a class that implements all methods of [CredentialsService interface](https://github.com/kaaproject/kaa/blob/1d429a30bb4b5206376b740bb21483929a881ace/server/node/src/main/java/org/kaaproject/kaa/server/node/service/credentials/CredentialsService.java).
```

Comment:
I don't like this link. Rather, stick to master branch links whenever possible (actually, let's add this to the Rules Of Conduct:):
https://github.com/kaaproject/kaa/blob/master/server/node/src/main/java/org/kaaproject/kaa/server/node/service/credentials/CredentialsService.java